### ISSUE TITLE
cppcheck flushes out a number of bits of code to look over

### ISSUE THREAD

**Issue by erich666** [syoyo/tinygltf#263]
Try the free [cppcheck](http://cppcheck.sourceforge.net/) tool on the code. 

![image](https://user-images.githubusercontent.com/803995/82598968-7c7aeb80-9b79-11ea-873f-e1d8e8f4a4e8.png)

and that's just the start. Most of these warnings can be ignored as "true but picky"or similar, but a few seem worthwhile. For example, cppcheck flags this line:

```if (req_comp && req_comp >= 3) // we can directly decode 3 or 4```

with:

>Redundant condition: If 'rec_comp >= 3', the comparison 'req_comp' is always true.

That is, the check of req_comp isn't needed, since the second condition is enough.

It's easy to run cppcheck, and the code will be just that little bit cleaner.

**Comment by erich666** [comment]
Here's the XML file (faked as a text file) when run on basic.sln, to save you having to install cppcheck.

[tinygltf-master.xml.txt](https://github.com/syoyo/tinygltf/files/4664527/tinygltf-master.xml.txt)

**Comment by syoyo** [comment]
> if (req_comp && req_comp >= 3) // we can directly decode 3 or 4

This code is located in `stb_image.h`, not TinyGLTF.

Please post cppcheck issue to stb project: https://github.com/nothings/stb

NOTE: tinygltf code itself is checked by clang  `-Weverything`, which may give better C++ syntax checking than cppcheck.

**Comment by erich666** [comment]
Thanks, good to know - I hadn't heard of -Weverything. And thanks for your diligence in keeping your own code base clean and tidy.

**Comment by erich666** [comment]
Oops, I closed this one in haste. Scrolling down in cppcheck, I see a bunch of warnings you might want to take 5 minutes zipping through - your choice. I mention these not to be "that guy," but rather that some of us love tinygltf here at NVIDIA and I like giving back, making it that 0.1% better.

![image](https://user-images.githubusercontent.com/803995/82693911-5b290680-9c30-11ea-95dc-4ea52b6d0f44.png)

The XML file I linked to before will show the full messages.
